src/Sinch.ServerSdk/Calling/Callbacks/Response/Internal/BridgedCallSvamletBuilders.cs

```diff
@@ -126,18 +126,37 @@ protected IConnectMxpSvamletResponse ConnectMxp(string userName, string defaultC
         }
 
         public ISvamletResponse RunMenu(string menuId, IMenuBuilder menuBuilder)
+        {
+            return RunMenu(menuId, false, menuBuilder);
+        }
+
+        public ISvamletResponse RunMenu(string menuId, bool enableVoice, IMenuBuilder menuBuilder)
         {
             var menus = menuBuilder.Build().ToDictionary(m => m.Id, m => m);
 
             if (!menus.ContainsKey(menuId))
                 throw new BuilderException("Menu '" + menuId + "' is not defined");
 
+            if (!enableVoice)
+            {
+                foreach(var menu in menus)
+                    if (menu.Value.Options != null)
+                        foreach (var option in menu.Value.Options)
+                        {
+                            if (option.Input != null && option.Digit == null)
+                                throw new BuilderException("Menus with voice option disabled only support DTMF input");
+
+                            option.Input = null;
```

Comment:
That's correct, but I think it's good to try to avoid sending extra bytes if we can. Alternatively, I would also nullify `option.Dtmf` when voice is enabled.

Reply:
I considered a case when user doesn't enable voice but provides dtmf option in `Input`. Won't it be lost if the property is nullified?

Reply:
In that case the check on line 146 will throw an exception.

Reply:
Why should we force client to use DTMF instead of more generic Input?

In the description you wrote:

> the input property matches both DTMF and voice inputs (afaik)

I also assumed that.

Reply:
It's a problem we're having at the moment with this nuget, which was originally intended to be used with the old platform. The old platform doesn't support `Input` - it's a new field we introduced for the voice ivrs. If we force the generated svaml to only define the `Input` field, we will break old functionality.

Reply:
Ok, I saw that in `MenuOption` constructor `input` argument is converted to DTMF (`TryMapToDtmf `), I thought it would solve the compatibility problem.